src/tools/tests/recipe2plan-test.ts

```diff
@@ -34,9 +37,93 @@ $ tools/update-goldens \n\n`
     assert.lengthOf(decoded['particleSpecs'], 3);
   }));
   it('filters generated plans by provided name', Flags.withDefaultReferenceMode(async () => {
-    const encoded = await recipe2plan(inputManifestPath, OutputFormat.Proto, 'Consumption') as Uint8Array;
+    const manifest = await Runtime.parseFile(inputManifestPath);
+    const encoded = await recipe2plan(manifest, OutputFormat.Proto, 'Consumption') as Uint8Array;
     const decoded = ManifestProto.decode(encoded);
     assert.lengthOf(decoded['recipes'], 1);
     assert.lengthOf(decoded['particleSpecs'], 1);
   }));
+  it('allows namespace to prefix particle path', Flags.withDefaultReferenceMode(async () => {
+    const manifest = await Runtime.parse(`
+    meta
+      namespace: arcs.core.data.testdata
+      
+    particle Writer in '.Writer'
```

Comment:
would it make sense to use `Writer` instead of `.Writer`?

Reply:
I thought about that. How would we differentiate between classes in the current directory with classes at the end of the path? 

The dot prefix is a simple way to disambiguate this. It also follows an Android Manifest convention, see [these docs](https://developer.android.com/guide/topics/manifest/manifest-intro#components).

Reply:
hmm... I think I mean to say "root directory", after reflection.

Reply:
Makes sense. TIL about android manifest conventions! :)